theano/gpuarray/multinomial.py

```diff
@@ -327,6 +327,7 @@ def gpu_kernels(self, node, name):
 
                     if (! %(replace)s )
                     {
+                        __syncthreads();
```

Comment:
I'm not sure this is good. Are you sure then when this is executed 100% of the threads in a given blocks execute this line?

Otherwise, there will be cases where we never finish.